src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -83,6 +83,11 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
+        List<Participation> participationList = personToEdit.getParticipation();
+        // UnEnroll student from current participation first
+        for (Participation p: participationList) {
+            new UnEnrollCommand(index, p.getTutorialSubject()).execute(model);
```

Comment:
This for loop might not work also because we are looping over each participation object, but we are removing it from the participation list at the same time. We should create a duplicate participation list and loop over it, so UnEnrollCommand will not be executing on this participation object in the for loop.

Reply:
hmm its working now for some reason. Lets flag it out like what jiehui suggested